incentivization.md

```diff
@@ -0,0 +1,342 @@
+
+Note also that there may be different market models that may motivate the choice of the payment method.
+One model assumes that each client pays for its own requests.
+Another model includes (centralized?) entities (i.e., developers of Waku-based apps) that pay for their users in bulk.
+
+We also note that:
+- eventually the protocol may support multiple payment methods;
+- however, the MVP version should be simple, which likely means supporting just one payment method;
+- if the initially supported payment method is an ERC-20 token, it should be easy to add other ERC-20 tokens later, including a potential WAKU token.
+
+# Evaluation
+
+We should think about what the success metrics for an incentivized protocol are, and how to measure them both in simulated settings, as well as in a live network.
+
+# Future work
+
+Let us now outline some of the open questions beyond MVP.
+## Price discovery
+
+To offer a reasonable price, a server should understand its costs.
+The costs of a Store server are storage, bandwidth, and computation.
+A Store server does two things: it stores messages, and serves messages to clients.
+
+The cost of storing messages is composed of:
+- storage:
+	- storage costs of all older messages: proportional to cumulative (message size x time stored);
+	- the cost for I/O operations for storing new messages (roughly constant per unit time, though may fluctuate due to caching, disk fragmentation, etc.);
+- bandwidth (download) for receiving new messages;
+- computational costs.
+
+The cost of serving messages to clients, per unit time, is composed of:
+- bandwidth
+	- upload: proportional to (number of clients) x (length of time frame requested) x (message size);
+	- download: proportional to the number of requests;
+- computational cost of handling requests.
+
+Storage is likely the dominating cost.
+Storage costs is proportional to the amount of information stored and the time it is stored for.
+A cumulative cost of storing a single message grows linearly with time.
+The number of messages in a response may be approximated by the length of the time frame requested.
+
+Computation: the server spends computing cycles while handling requests.
+This costs likely depends not only on the computation itself, but also at the database structure.
+For example, retrieving old or rarely requested messages from the local database may be more expensive than fresh or popular ones due to caching.
+
+More formal calculations should be done, under certain assumptions about message flow (i.e., that it is constant).
+
+## Price negotiation
+
+If the server offers the price that is too high for the client, the client has no means to make a counter-offer.
+This results in wasted bandwidth on requests that don't result in responses.
+We could introduce a price negotiation step in the protocol, where the client and the server would exchange messages naming their acceptable prices until they agree of one of them decides to stop the negotiation.
```

Comment:
Perhaps a small trivial payment is made to request a quote for message retrieval, with the idea being that the client sends requests to multiple servers, with the savings by finding the lowest price server exceeding the cost of sending multiple requests.

Reply:
Extracted the idea to #50.